### PR TITLE
Show delete button for usercreated functions

### PR DESCRIPTION
**Pull request by balderdash**
This also hides the delete button on prewritten functions in CSinA levels. Based on the levels I looked at, deleting a prewritten function just makes it harder/impossible to finish the level without starting over, so I'm happy with that change.

### CODE REVIEW

File: blockly-core/core/ui/function_editor.js [hunk 1]
```diff
@@ -798,6 +799,20 @@ Blockly.FunctionEditor.prototype.addDeleteButton_ = function () {
...
 
+/**
+ * Show or hide delete button. If none exists and it should be shown, create
+ * it.
+ */
+Blockly.FunctionEditor.prototype.setDeleteButtonVisible_ = function(visible) {
+  if (this.deleteButton_) {
+    this.deleteButton_.setVisible(visible);
+  } else if (visible) {
+    this.hasDeleteButton = true;
```
Comment (Hamms) [inline blockly-core/core/ui/function_editor.js]: this doesn't appear to be used

Comment (Hamms) [inline blockly-core/core/ui/function_editor.js]: Ah, I see it's used by `addDeleteButton`. Is this set anywhere else? Why is it not set by `addDeleteButton`?

File: blockly-core/tests/modal_editors_test.js [hunk 2]
```diff
@@ -120,6 +134,28 @@ function test_initializeFunctionEditor() {
...
+}
+
+function test_functionEditor_deleteButton() {
+  var container = Blockly.Test.initializeBlockSpaceEditor();
+  initializeFunctionEditor(USER_CREATED_PROCEDURE);
+  openFunctionEditor('test-usercreated-function');
+
+  assertNotNull('Function exists',
+      Blockly.mainBlockSpace.findFunction('test-usercreated-function'));
+  assertEquals('Delete button says "Delete"', 'Delete',
```
Comment (Hamms) [inline blockly-core/tests/modal_editors_test.js:153]: probably want to throw in a "Function editor has delete button" test before this; if something broke and the delete button doesn't exist, this test will probably raise something like a `TypeError: Cannot read property 'textContent' of undefined` error, and the helper message will be confusing.

**Comment by Hamms** [comment]
Couple of comments, but otherwise LGTM! My apologies for the delay in reviewing.

File: blockly-core/core/ui/function_editor.js (hunk 1 above)
Comment (balderdash) [inline blockly-core/core/ui/function_editor.js]: Yeah, turns out `hasDeleteButton` was no longer needed. It was just being used to differentiate between the function editor and the contract editor (showing the delete button only in the latter), but they now have the same behavior.

Removed it.

File: blockly-core/tests/modal_editors_test.js (hunk 2 above)
Comment (balderdash) [inline blockly-core/tests/modal_editors_test.js:153]: Sounds good, done.

**Pull request merged**